src/Compilers/Core/Portable/Symbols/ISymbolExtensions_IsAccessibleWithin.cs

```diff
@@ -0,0 +1,348 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    public static partial class ISymbolExtensions
+    {
+        /// <summary>
+        /// Checks if 'symbol' is accessible from within 'within', with
+        /// an optional qualifier of type 'throughTypeOpt' to be used to resolve
+        /// protected access.
+        /// </summary>
+        public static bool IsAccessibleWithin(
```

Comment:
>IsAccessibleWithin [](start = 27, length = 18)

Given the many "interesting" and controversial qualities of this API, I do not believe this API belongs in CodeAnalysis assembly as an extension method with such an innocent name. Compilers should not be using it under no circumstances in their internal implementation and since all symbols implement ISymbol, it is very easy to fall into this trap. Ideally this API should be pulled out into an assembly, which is not referenced by assemblies in the compiler layer, including this one.

Reply:
Added an Obsolete instance method to Symbol to interfere with the use of this extension method in the compilers.

---
In reply to: [185284396](https://github.com/dotnet/roslyn/pull/26367#discussion_r185284396) [](ancestors = 185284396)

Reply:
> Added an Obsolete instance method to Symbol to interfere with the use of this extension method in the compilers.

This is not sufficient, sometimes compilers operate with interfaces

---
In reply to: [185310868](https://github.com/dotnet/roslyn/pull/26367#discussion_r185310868) [](ancestors = 185310868,185284396)

Reply:
I am not aware of any scenarios where the compilers use interfaces when performing access checking.

---
In reply to: [185311211](https://github.com/dotnet/roslyn/pull/26367#discussion_r185311211) [](ancestors = 185311211,185310868,185284396)

Reply:
> I am not aware of any scenarios where the compilers use interfaces when performing access checking.

My concern was not limited to current code, and it is no addressed by your recent changes. Please do not mark this thread as resolved bases on that.

---
In reply to: [185311434](https://github.com/dotnet/roslyn/pull/26367#discussion_r185311434) [](ancestors = 185311434,185311211,185310868,185284396)